thingsboard_gateway/connectors/ftp/backward_compatibility_adapter.py

```diff
@@ -0,0 +1,93 @@
+from copy import deepcopy
+import json
+
+
+class FTPBackwardCompatibilityAdapter:
+
+    def __init__(self, config: dict) -> None:
+        self._config = deepcopy(config)
+        self._converted_config = {
+            "parameters": {
+            },
+            "paths": [],
+            "requestsMapping": {
+                "attributeUpdates": [],
+                "serverSideRpc": []
+            }
+        }
+
+    def convert(self) -> dict:
+
+        self._converted_config['parameters']['host'] = self._config.get('host')
+        self._converted_config['parameters']['port'] = self._config.get('port')
+        self._converted_config["parameters"]["TLSSupport"] = self._config.get("TLSSupport", False)
+        security = self._config.get('security', {})
+        self._converted_config['parameters']['security'] = security
+        self._converted_config["parameters"]["security"]["type"] = security.get("type")
+        self._converted_config["parameters"]["security"]["username"] = security.get("username")
+        self._converted_config["parameters"]["security"]["password"] = security.get("password")
+        old_paths = self._config.get('paths', [])
+        new_paths = self._convert_path_entries(old_paths)
+        self._converted_config["paths"] = new_paths
+        self._converted_config["requestsMapping"]["attributeUpdates"] = self._config.get("attributeUpdates", [])
+        self._converted_config["requestsMapping"]["serverSideRpc"] = self._config.get("serverSideRpc", [])
+
+        return self._converted_config
+
+    @staticmethod
+    def _convert_path_entries(old_paths: list) -> list:
+        new_paths = []
+        for old_path in old_paths:
+            new_path_data = FTPBackwardCompatibilityAdapter._convert_path(old_path)
+            new_paths.append(new_path_data)
+        return new_paths
+
+    @staticmethod
+    def _convert_path(old_path_data: dict) -> dict:
+        new_path_data = old_path_data
```

Comment:
It may be better to use additional dict to collect converted configuration.

It gives ability to avoid usages of deepcopy in the future for internal objects.